### ISSUE TITLE
v0

### ISSUE THREAD

**Issue by U-C-S** [U-C-S/Hurl#153]

**Comment by Jay-o-Way** [comment]
May I suggest:

- include the BrowserSelector page in the main window (for quick access to settings, remove `wpfui`, theme adapting)
  - Brush up the XAML for that page (let me know if you want me to pick up something here)
- use a (compact) [NavigationView](winui3gallery://item/NavigationView) control
- use as native as possible solution for TitleBar

**Comment by U-C-S** [comment]
> include the BrowserSelector page in the main window

Not sure what you mean by this. 

> use a (compact) NavigationView control

I don't prefer the compact one, but something like MS Store or Windows Settings is preferable. 
(the current implementation in Hurl Settings doesn't clean up memory, so memory usage rises up in 1 MBs as it's navigated, even to the same page)

**Comment by Jay-o-Way** [comment]
> include the BrowserSelector page in the main window

My idea would be to combine this page with the rest
![image](https://github.com/user-attachments/assets/c423e3b8-109a-42bb-813a-72f5a42369d7)

Then the Nav button/menu can be compact or hidden to keep UI clean.

**Comment by U-C-S** [comment]
I personally prefer to keep Hurl Settings and Selector separately.

**Comment by Jay-o-Way** [comment]
📰 WASDK 1.6.0 is out 😄

**Comment by Jay-o-Way** [comment]
Any specific areas / issue you want help with?

**Comment by U-C-S** [comment]
> Any specific areas / issue you want help 

Guess so, one thing I want to experiment is migrate `Hurl.BrowserSelector` (maybe should be renamed to `Hurl.Selector`) to WinAppSDK. so only thing I am not sure there is how minimizing to Tray works there.

So, If you are interested, A minimal PR that only does migration to WinAppSdk and WinUi components without much UI or functional changes or you can choose to help with anything else, but focused PRs haha.

**Comment by Jay-o-Way** [comment]
> I want to experiment migrate `Hurl.BrowserSelector` to WinAppSDK

Hah, that's a big issue and not easy. I think you have more experience with this than I do, because you already worked on the Settings project. I think I'll try to convert the ListViews to Navigation control first :)

**Comment by U-C-S** [comment]
Waiting on https://github.com/actions/runner-images/issues/10734 to release [`v0.9.1`](https://github.com/U-C-S/Hurl/compare/v0.9.0...v0.9.1)

https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#workloads-components-and-extensions

**Comment by uqiu** [comment]
too many thanks for your amazing work.

**Comment by U-C-S** [comment]
`v0.10` Scope Revaluation

- [ ] Move Hurl.Settings to possible upcoming winappsdk `1.7` (for now, will move it `1.6`)
- [ ] Update to .NET 9
- [ ] Use learnings from #174 to refactor Hurl.Settings app
- [ ] ..... (TODO: add more to the list)

All of the features originally planned will now move for a next release instead. Why? The way the current code is written doesn't support adding new features without a significant refactor yet again as I learnt from while developing #174. 
Additionally I am trying to write a new library for Glob patterns for URLs with some ideas I had ([see the project here](https://github.com/U-C-S/uri-globbing), it's still a WIP).

**Comment by Jay-o-Way** [comment]
Yeah, at some point (if it were me) I would start thinking about starting over from (almost) scratch, using the latest. That needs a huge amount of time and effort, but does mean a fresh clean code, and being able to include/solve many ideas and issues.

P.S. To me, Regular Expressions are familiar, but Glob patterns are not. To me, they seem like the same thing. What is your experience with Glob and is it worth supporting both?

Also, **1.7 Preview 1** is out 🚀 https://github.com/microsoft/windowsappsdk/releases

**Comment by NSurtsev** [comment]
"As we only support win11"
Hey, wait a second. I'm on Win10 and lonterm support like at least for 3 years.

**Comment by U-C-S** [comment]
[Support from microsoft for Win10 ended on Oct 2025](https://support.microsoft.com/en-us/windows/windows-10-support-has-ended-on-october-14-2025-2ca8b313-1946-43d3-b55c-2b95b107f281).

@NSurtsev Though Hurl might keep working for Windows 10 but I don't have testing bandwidth to handle Windows 10 right now. I wanted to move to more newer UI frameworks and tooling, to further ease my development efforts and I dont think they support windows 10 anymore either. Just a sacrifice I had to make unfortunately 😕.

Also in the above context, "As we only support win11", I meant to say remove some settings (like Solid background, No White Border) that I made with windows 10 in mind.

---

Actually, Give the [newest release 0.10-alpha-2](https://github.com/U-C-S/Hurl/releases/tag/v0.10.0-alpha-2) a try, if you are using win10 and maybe lemme know here.

**Comment by NSurtsev** [comment]
It's ok if UI is different style.
I'm interested only in core functionality.

**Comment by U-C-S** [comment]
<img width="2137" height="1187" alt="Image" src="https://github.com/user-attachments/assets/8b2c1699-5d8f-493f-a567-6747d5e6b512" />

---

QuickView Windows, inspired out from [Little Arcs](https://resources.arc.net/hc/en-us/articles/19235387524503-Little-Arc-Quick-Lookups-Instant-Triaging) and Powered by MSEdge WebView2 (WIP)

**Comment by Jay-o-Way** [comment]
If it were me, i would say _target_ the latest sdk but _support_ the latest build of Windows 10 (and later).

**Comment by NSurtsev** [comment]
Updated from latest stable (0.9.3), but app doesn't work.
Uninstalled and installed the app. Still does not.
Uninstalled and reinstalled stable. It works.

P.S. I am on Win10. If it's the case, I'll just stay on 0.9.3.

**Comment by gtflip** [comment]
I'm on Windows 10.  I tried installing it, but it didn't work (didn't pop up at all).  So, I uninstalled it and deleted all it's files (and maybe registry settings?).  It's working now (0.10.0-alpha-2) and definitely more reliable on popping up when I open a link from another program.  👍 

I noticed that the link on the About page for "Source Code - GitHub Repository" seems to be blank.  I can open an issue if you'd rather me do it that way.

**Comment by sjain882** [comment]
Just checking, is it already possible or intended to be possible to:

- Have certain browsers **hidden** by default, but only **show** during certain URL/rule matches
- Have certain browsers **visible** by default, but only **hide** during certain URL/rule matches

in v0.10+?

Thanks

**Comment by U-C-S** [comment]
> * Have certain browsers **hidden** by default, but only **show** during certain URL/rule matches
> * Have certain browsers **visible** by default, but only **hide** during certain URL/rule matches

Not possible. but intended to be soon.
